libr/util/str.c

```diff
@@ -2928,3 +2929,21 @@ R_API char* r_str_highlight(char *str, const char *word, const char *color) {
 	}
 	return strdup (o);
 }
+
+R_API int r_str_scpy (char *dst, const char *src, int count) {
+  if (count <= 0) {
+    return -E2BIG;
+  }
+  int ret = 0;
+  while (count) {
+    if (!(dst[ret] = src[ret])) {
+      return ret;
+    }
+    ret++;
+    count--;
+  }
+  if (ret) {
+    dst[ret-1] = '\0';
+  }
+  return -E2BIG;
```

Comment:
returning error usually means no data has been copied, but the buffer has been modified at this point, so what about returning the amount of bytes copied like strncpy does